### ISSUE TITLE
Trying to reproduce results of the paper

### ISSUE THREAD

**Issue by rielmvp** [PKU-ML/adainf#4]
Thank you for the hard work, and especially for the organized repository enabling me to try to reproduce results. I would like to inquire about some difficulties I'm having to reproduce the paper, especially the linear probing/offline evaluation results. I am trying to reproduce the results you've shared in the paper, specifically for the linear probing/offline evaluation results of SimCLR on CIFAR-100.

![Image](https://github.com/user-attachments/assets/9de4aba2-756a-442e-894a-47ec6df1975d)

For the Table 1 results on CIFAR-100, it seems that it is the result of linear probing/offline evaluation, which in solo-learn would mean running "main_linear.py" with the .yaml file for linear probing for ImageNet-100 in your repository. In short, I haven't been able to replicate the results for linear probing/offline evaluation (first figure below) though I have been able to run the pretraining script and get online evaluation numbers that are similar to the linear probing numbers that were reported in the paper (second figure below).

![Image](https://github.com/user-attachments/assets/d07add4c-388d-4b37-a78c-4090e156598c)

![Image](https://github.com/user-attachments/assets/9e4ec5b5-d4ad-4652-9315-ebf2ab726a7f)

Apologies if I misunderstood the results that were posted in the paper and the differences in online evaluation linear probing/offline evaluation. I myself am new to researching self-supervised learning and using the solo-learn library. I would really appreciate it if you would let me know any misunderstandings I made in interpreting or reproducing the paper's results.

Thank you very much beforehand and best regards,

**Comment by zjizhe** [comment]
Thanks for your question. Due to a busy schedule today, I’d appreciate it if I could provide you with a detailed response by tomorrow.

**Comment by rielmvp** [comment]
That would be great, thank you very much! Any detailed response in your convenient time would be very helpful.

**Comment by zjizhe** [comment]
Hi. When conducting offline linear evaluation on SimCLR trained on CIFAR100 using linear/imagenet-100/simclr.yaml, please make sure to set ”auto_augment“ to "False". This is because ”auto_augment“ is an augmentation designed for ImageNet and is not suitable for CIFAR. If you have any other questions, please feel free to ask.

**Comment by rielmvp** [comment]
Thanks for the response. I got results similar to the paper when I conducted offline evaluation with setting "auto_augment" to "False" in the given script, so I was able to reproduce CIFAR100 results:

![Image](https://github.com/user-attachments/assets/86eb44be-598a-433a-9240-a01d16af5c22)

Is it correct to assume that the results in the paper below (as shown in Tables a and b, and Figure 6a) are results of offline linear evaluation of 100 epochs, while Figure 6b shows the result of online linear evaluation along the 500k training process? I'm not really sure but it seems that offline linear evaluation tends to give bit higher accuracy compared to evaluated online (as seen in the figures). Is this true generally in self-supervised learning? Also, besides "auto-augment" in this case, will changing other parameters of the script (such as optimizer or scheduler) greatly affect offline linear evaluation accuracy?

![Image](https://github.com/user-attachments/assets/9b51c639-f5d1-4dd1-ac82-cccb7e84f65f)

Thank you again~

**Comment by zjizhe** [comment]
In our paper, the reported results are based on online linear evaluation. Theoretically, the results of offline linear accuracy are expected to surpass those of online linear evaluation. This is because, during offline linear evaluation, the parameters of the backbone have already been optimized and remain unchanged. Regarding other parameters in the offline evaluation, we have not conducted extensive experimentation.
